packages/core/core.ts

```diff
@@ -316,7 +356,52 @@ async function loadFuzzFunction(options: Options): Promise<fuzzer.FuzzTarget> {
 			`${options.fuzzTarget} does not export function "${options.fuzzEntryPoint}"`
 		);
 	}
-	return fuzzFn;
+	return wrapFuzzFunctionForBugDetection(fuzzFn);
+}
+
+/**
+ * Wraps the given fuzz target function to handle exceptions from both the fuzz target and bug detectors.
+ * Ensures that exceptions thrown by bug detectors have higher priority than exceptions in the fuzz target.
+ */
+export function wrapFuzzFunctionForBugDetection(
+	originalFuzzFn: fuzzer.FuzzTarget
+): fuzzer.FuzzTarget {
+	if (originalFuzzFn.length === 1) {
+		return (data: Buffer): void | Promise<void> => {
+			let fuzzTargetError: unknown;
+			let result: void | Promise<void>;
+			try {
+				result = (originalFuzzFn as fuzzer.FuzzTargetAsyncOrValue)(data);
+			} catch (e) {
+				fuzzTargetError = e;
+			}
+			return handleErrors(result, fuzzTargetError);
+		};
+	} else {
+		return (data: Buffer, done: (err?: Error) => void): void => {
+			let fuzzTargetError: unknown;
+			try {
+				originalFuzzFn(data, (err?: Error) => {
+					done(getFirstFinding() ?? err);
```

Comment:
Clean `firstFinding`, similar to the code in `handleErrors`?

Reply:
Good catch! It seems, I didn't write enough tests.